src/mame/misc/mjsenpu.cpp

```diff
@@ -205,43 +184,45 @@ uint32_t mjsenpu_state::muxed_inputs_r()
 		break;
 
 	case 0x9e:
-		return ioport("MUX_9E")->read();
+		return m_mux_9e->read();
 
 	case 0x9d:
-		return ioport("MUX_9D")->read();
+		return m_mux_9d->read();
 
 	case 0x9b:
-		return ioport("MUX_9B")->read();
+		return m_mux_9b->read();
 
 	case 0x97:
-		return ioport("MUX_97")->read();
+		return m_mux_97->read();
 
 	case 0x8f:
-		return ioport("MUX_8F")->read();
+		return m_mux_8f->read();
 	}
```

Comment:
This isn’t a multiplexer – it’s a switch matrix, using the low five bits to select rows.  Could you please:
* Use a `required_ioport_array` for the five mahjong keyboard rows
* Convert this function to read the rows as a switch matrix

You can find examples of switch matrix input in many mahjong drivers.  The code should end up something like:
```c++
	uint32_t result = 0xffffffff;
	for (unsigned i = 0; m_matrix_inputs.size() > i; ++i)
	{
		if (!BIT(m_mux, i))
			result &= m_matrix_inputs[i]->read();
	}
	return result;
```